File: messageGo/grpc/pubsubEventData/mint_nft.go
package pubsubEventData

import (
	base_data "game-message-core/grpc/baseData"
)

type MintNftInfo struct {
	Owner      int64                        `json:"owner"`
	Items      []base_data.GrpcItemBaseInfo `json:"items"`
	FromEntity base_data.EntityInfo         `json:"fromEntity"`
}

func (p *MintNftInfo) Clear() {
	p.Owner = 0
	p.FromEntity.Clear()
	if p.Items != nil {
		p.Items = p.Items[:0]
	}
}

// 通知mainService make User nft input
type MintNftEvent struct {
	MsgVersion  int64                 `json:"msgVersion"` // 消息版本号 值为毫秒时间戳
	FormService base_data.ServiceData `json:"formService"`
	Nft         MintNftInfo           `json:"nft"`
}

func (p *MintNftEvent) Clear() {
	p.MsgVersion = 0
	p.FormService.Clear()
	p.Nft.Clear()
}

// 通知mainService 批量 make User nft input
type MultiMintNftEvent struct {
	MsgVersion  int64                 `json:"msgVersion"` // 消息版本号 值为毫秒时间戳
	FormService base_data.ServiceData `json:"formService"`
	Nfts        []MintNftInfo         `json:"nfts"`
}

func (p *MultiMintNftEvent) Clear() {
	p.FormService.Clear()
	for i, _ := range p.Nfts {
		p.Nfts[i].Clear()
	}
	if p.Nfts != nil {
		p.Nfts = p.Nfts[:0]
	}
}

File: messageDotNet/grpc/EventData/MintNft.cs
using System;
using UnityEngine;


[Serializable]
public class MintNftInfo
{
    public long Owner;
    public GrpcItemBaseInfo[] Items;
    public EntityInfo FromEntity;
}

/// <summary>
/// dapr call mainService mint User nft input
/// </summary>
[Serializable]
public class MintNftEvent
{
    // 消息版本号 值为毫秒时间戳
    public long MsgVersion;
    public ServiceData FormService;

    public MintNftInfo Nft;
}

/// <summary>
/// dapr call mainService mint User nft input
/// </summary>
[Serializable]
public class MultiMintNftEvent
{
    // 消息版本号 值为毫秒时间戳
    public long MsgVersion;
    public ServiceData FormService;
    public MintNftInfo[] Nfts;
}